src/wrapper_functions.cc

```diff
@@ -172,17 +172,17 @@ LibGitReference branch_lookup(git_repository* repo, const std::string& branch_na
     return { ref, git_reference_free };
 }
 
-// std::string branch_remote_name(git_repository* repo, const std::string& branch_name)
-// {
-//     git_buf buf;
-//     if (git_branch_remote_name(&buf, repo, branch_name.c_str()))
-//     {
-//         // gul14::cat("branch_remote_name: ", git_error_last()->message);
-//         return "";
-//     }
-//     auto ret = std::string{ buf.ptr };
-//     git_buf_dispose(&buf);
-//     return ret;
-// }
+std::string branch_remote_name(git_repository* repo, const std::string& branch_name)
+{
+    git_buf buf{ };
+    auto result = git_branch_remote_name(&buf, repo, branch_name.c_str());
+    if (result) {
+        git_buf_dispose(&buf);
```

Comment:
It might be better to put the `git_buf_dispose()` into a `gul14::finally()` cleanup object.

Reply:
Right.

Reply:
> Also added code to free the buffer in case of a failure, as the docu does not say if or if not the buffer might get filled in that case.

Checked the code, `git_branch_remote_name` is in fact macro magick :-(
But we indeed need to free it always.

Implemented via finalizer.